src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -1128,6 +1172,9 @@ static int netlink_connect(Config *cfg, struct nl_sock *sock, int nl_id, int fd,
   NLA_PUT_U64(msg, NBD_ATTR_SIZE_BYTES, size);
   NLA_PUT_U64(msg, NBD_ATTR_BLOCK_SIZE_BYTES, RBD_NBD_BLKSIZE);
   NLA_PUT_U64(msg, NBD_ATTR_SERVER_FLAGS, flags);
+  if (cfg->persist) {
+    NLA_PUT_U64(msg, NBD_ATTR_DEAD_CONN_TIMEOUT, cfg->persist_timeout);
+  }
```

Comment:
For the `NBD_ATTR_DEAD_CONN_TIMEOUT` we'd better to set it normally for all the devices defautly. This is not only useful for the remap but also for multiple connections cases.

We can just set one default value, like 30s as default, and keep the `--persist_timeout` options to allow user to disable it or change it.

Reply:
rbd-nbd doesn't support multiple connections right now, so in this case it is useful for remap only.

I don't mind to always set `DEAD_CONN_TIMEOUT` but I don't like very much that `persist_timeout` option will be used for this, even if it is not `persist` mode, which may be a little confusing for users.

So I would either leave it as it is now (the timeout is set only if `--persist` is on) or change `persist_timeout` name to something more generic, still the name should suggest that it is useful for `persist` mode. I can't think anything better than just `--dead-connection-timeout`, though I am not like this very much.

Reply:
Shouldn't `--persist` just be the default? We have the `rbd-nbd unmap` command which can be used to cleanly shut down a device (or cleanup a device from a crashed `rbd-nbd` process).

Reply:
Ack. I will try.

Reply:
That's okay for me keept the `--persist_timeout` as it is is now.

Maybe if we will set the `persist` is as default, shouldn't the `timeout` also has one default value ? Or you will also randomly could see the IO error before the new socket connection is set.

This timeout is trying to not fail the IO cmds immediately if they see all the connections are dead before their own timer expires.